apps/clearance_ui/src/components/clearance_library/bulk_conclusions/ActionCell.tsx

```diff
@@ -2,28 +2,169 @@
+                },
+                params: {
+                    id: row.original.id,
+                },
+            },
+            {
+                onSuccess: () => {
+                    queryClient.invalidateQueries(bcKey);
+
+                    meta?.setEditedRows(() => {
+                        const old = meta.editedRows;
+                        return {
+                            ...old,
+                            [parseInt(row.id)]: !old[parseInt(row.id)],
+                        };
+                    });
+
+                    toast({
+                        title: "Bulk conclusion updated",
+                        description: "Bulk conclusion updated successfully",
+                    });
+                },
+                onError: (error) => {
+                    const msg = isAxiosError(error)
+                        ? error.response?.data.message
+                        : error.message;
+
+                    toast({
+                        variant: "destructive",
+                        title: "Error",
+                        description: msg,
+                    });
+                },
+            },
+        );
+
+    const setEditedRows = (e: React.MouseEvent<HTMLButtonElement>) => {
+        const elName = e.currentTarget.name;
+        if (e.currentTarget.name !== "save") {
+            meta?.setEditedRows(() => {
+                const old = meta.editedRows;
+                return {
+                    ...old,
+                    [parseInt(row.id)]: !old[parseInt(row.id)],
+                };
+            });
+        }
+
+        if (elName !== "edit") {
+            meta?.revertData(row.index, e.currentTarget.name === "cancel");
+            if (elName === "save") {
+                const isValidObj = isValidConcludedExpression(
+                    row.getValue("concludedLicenseExpressionSPDX"),
+                );
+
+                if (isValidObj.isValid) {
+                    const body: BulkConclusionUpdateData = {
+                        pattern: row.getValue("pattern"),
+                        concludedLicenseExpressionSPDX: row.getValue(
+                            "concludedLicenseExpressionSPDX",
+                        ),
+                        comment: row.getValue("comment"),
+                        local: row.getValue("local"),
+                    };
+
+                    UpdateBulkConclusion(body);
+                } else {
+                    let description = "Invalid concluded license expression";
+                    if (isValidObj.errWord !== null) {
+                        description +=
+                            ": syntax error in '" + isValidObj.errWord + "'";
+                    }
+                    toast({
+                        variant: "destructive",
+                        title: "Error",
+                        description: description,
+                    });
+                }
+            }
+        }
+    };
```

Comment:
What's the reason for this being one function with the `elName` check, rather than a handler for edit, save and cancel I believe? If sharing the function between them is not strictly needed, this could be quite a lot easier to follow with separate handlers.

Reply:
Looking back at the original example I used when implementing this for the license conclusions table, the function is quite a lot simpler in that example, but agree, here it is quite unreadable so separated it to different handlers.